packager/react-packager/src/DependencyResolver/Module.js

```diff
@@ -81,8 +82,11 @@ class Module {
     if (!this._reading) {
       this._reading = this._fastfs.readFile(this.path).then(content => {
         const data = {};
+
+        // Set an id on the module if it's using @providesModule syntax
         const moduleDocBlock = docblock.parseAsObject(content);
-        if (moduleDocBlock.providesModule || moduleDocBlock.provides) {
+        if (!this._depGraphHelpers.isNodeModulesDir(this.path) &&
```

Comment:
Maybe we should add a comment explaining why checking for this is necessary. It's not immediately obvious to me